library/cipher.c

```diff
@@ -976,6 +1006,14 @@ int mbedtls_cipher_crypt( mbedtls_cipher_context_t *ctx,
     int ret;
     size_t finish_olen;
 
+    CIPHER_VALIDATE_RET( ctx != NULL );
+    CIPHER_VALIDATE_RET( iv_len == 0 || iv != NULL );
+    CIPHER_VALIDATE_RET( ilen == 0 || input != NULL );
+    CIPHER_VALIDATE_RET( output != NULL );
+    CIPHER_VALIDATE_RET( olen != NULL );
+    if( ctx->cipher_info == NULL )
```

Comment:
Why did we introduce this check?

Reply:
I actually added the ckech in e768b94e86c4703a578afd071b7f364b69de6ba1 , and I can see there were no checks present at that point. In the recent commit I only made this unconditional, but given the context, I may as well remove it.

Reply:
Yes, please remove it, as it seems that the check wasn't there before and isn't needed here as far as I can see.